# FUNCIONES BASICAS DE *PYTHON*

python viene con un conjunto de funciones basicas, por ejemplo:

- print()
- help()
- type()
- len()
- sum()
- round()
- abs()
- max()
- min()
- str()
- int()
- float()
- input()
- tuple()
- list()
- set()
- range()
- dict()


## Parametros de un función

Si una función se invoca sin especificar algún parámetro esta puede trabajar con uno *por default.*

Por ejemplo, la función **print** tiene un parámetro *end* para especificar el carácter de fin de linea, que por default es un salto de linea.

In [56]:
print("HOLA")
print("Buenos dias ", "mundo")

HOLA
Buenos dias  mundo


Podemos especificar que tenga un caracter de tabulado del siguiente modo:

In [60]:
print("HOLA", end= " ")
print("Buenos dias ", "mundo")

HOLAkkkkBuenos dias  mundo


In [61]:
print("Buenos dias", "mundo", sep=' = ')

Buenos dias = mundo


In [5]:
print("Buenos dias", "mundo", sep=' + ', end="\n\t")

print("Hoy es miercoles 1ro", "de noviembre", sep=' : ')

Buenos dias + mundo
	Hoy es miercoles 1ro : de noviembre


In [62]:
help(print)

Help on built-in function print in module builtins:

print(...)
    print(value, ..., sep=' ', end='\n', file=sys.stdout, flush=False)
    
    Prints the values to a stream, or to sys.stdout by default.
    Optional keyword arguments:
    file:  a file-like object (stream); defaults to the current sys.stdout.
    sep:   string inserted between values, default a space.
    end:   string appended after the last value, default a newline.
    flush: whether to forcibly flush the stream.



# EJEMPLOS

# print()

In [ ]:
print("HOLA")

nombre = 'Enrique'

print("Hi", nombre)

HOLA
Hi Enrique


In [7]:
nombre = 'Enrique'
print("Hoy es lunes, me llamo", nombre, "Estoy con ganas de trabajar")

Hoy es lunes, me llamo Enrique Estoy con ganas de trabajar


In [63]:
nombre = 'Enrique'
print(f"Hoy es lunes, me llamo {nombre}. Estoy con ganas de trabajar")

Hoy es lunes, me llamo Enrique. Estoy con ganas de trabajar


In [64]:
nombre = 'Enrique'
print("Hoy es lunes, me llamo {0}. Estoy con ganas de trabajar".format(nombre))

Hoy es lunes, me llamo Enrique. Estoy con ganas de trabajar


# len()
Para averiguar el tamano de una coleccion, por ejemplo, una lista.

In [10]:
lista1 = ['a', 'b', 'c', 'x']

longitud_lista1 = len(lista1)
print ("Longitud de la lista1 :", longitud_lista1 )

nombre = 'T & C Week'
print ("Longitud de \'T & C Week\'  :", len(nombre) )

Longitud de la lista1 : 4
Longitud de 'T & C Week'  : 10


In [65]:
tup1 = ('a', 'b', 'c', 'xx', 13)

longitud_tup1 = len(tup1)
print ("Longitud de tup1 :", longitud_tup1 )

Longitud de tup1 : 5


In [ ]:
c1 = {'a', 'b', 'c', 'a', 'a'}

print ("Longitud de c1 :", len(c1) )

Longitud de c1 : 3


# type()

Averigua el tipo de dato.

In [13]:
nombre = 'TECSUP'
cantidad = 100.0
lista1 = ['a', 'b', 'c']

print("Tipo de dato de nombre : ", type(nombre) ) # string
print("Tipo de dato cantidad: ", type(cantidad) ) # real
print("Tipo de dato : ", type(lista1) )

Tipo de dato de nombre :  <class 'str'>
Tipo de dato cantidad:  <class 'float'>
Tipo de dato :  <class 'list'>


In [15]:
c1 = {'a', 'b', 'c'}
d1 = {'a':10, 'b':100, 'c':2000}

print("Tipo de dato de c1 : ", type(c1) ) # string
print("Tipo de dato de d1 : ", type(d1) ) # string


Tipo de dato de c1 :  <class 'set'>
Tipo de dato de d1 :  <class 'dict'>


In [67]:
x = True
print(type(x))

y = (18, 19, 16)
print(type(y))

<class 'bool'>
<class 'tuple'>


# sum()

In [ ]:
lista2 = [11, 12, 13, 14, 15]
tupla = (1, 2, 3)

conjunto = {2, 3, 3}

print("Suma de elementos de lista2:", sum(lista2) )
print("Suma de elementos de tupla:",sum(tupla))
print("Suma de elementos de variable \'conjunto\' :",sum(conjunto))

Suma de elementos de lista2: 65
Suma de elementos de tupla: 6
Suma de elementos de variable 'conjunto' : 5


In [ ]:
parte1 = (14,	16,	14,	15)
parte2 = (0,	18,	4,	19)

print("Suma de notas parte1:", sum(parte1) /4)
print("Suma de notas parte2:", sum(parte2) /4)

Suma de notas parte1: 14.75
Suma de notas parte2: 10.25


# round()

Redondea numeros reales.

In [75]:
medida = 12.51255

print(round(medida, 4))

12.5125


In [17]:
help(round)

Help on built-in function round in module builtins:

round(number, ndigits=None)
    Round a number to a given precision in decimal digits.
    
    The return value is an integer if ndigits is omitted or None.  Otherwise
    the return value has the same type as the number.  ndigits may be negative.



In [ ]:
calculo = 47/3

print(round(calculo, 0))

16.0


In [76]:
parte1 = (14,	16,	14,	15)
parte2 = (0,	18,	4,	19)

prom1 = sum(parte1) /4
prom2 = sum(parte2) /4
print("Suma de notas parte1:", round(prom1, 1) )
print("Suma de notas parte2:", round(prom2, 1) )

Suma de notas parte1: 14.8
Suma de notas parte2: 10.2


In [ ]:
help(round)

Help on built-in function round in module builtins:

round(number, ndigits=None)
    Round a number to a given precision in decimal digits.
    
    The return value is an integer if ndigits is omitted or None.  Otherwise
    the return value has the same type as the number.  ndigits may be negative.



# abs()

Obtiene el valor absoluto.

In [ ]:
valor = 10 - 20
print( valor )
print( abs(valor) )

-10
10


# max()

Halla el valor maximo de una coleccion.

In [ ]:
lista2 = [0.5, 12.1, 3, 4, 5.5]
print("El valor maximo de la lista es", max(lista2) )

El valor maximo de la lista es 12.1


# min()

Halla el valor minimo de una coleccion.

In [ ]:
lista2 = [18, 12, 0, 20, 19]

print("La nota minima es", min(lista2) )

La nota minima es 0


Se desea obtener el promedio de las 4 notas mas altas de lista2.

In [78]:
lista2 = [18, 12, 0, 20, 19]
nota_baja = min(lista2)
print("Nota mas baja:", nota_baja)

suma_notas = sum(lista2)
print("Suma de todas las notas:", suma_notas)
suma_menos_nota_baja = suma_notas - nota_baja

print("Suma de todas las notas MENOS la nota mas baja:", suma_menos_nota_baja)
promedio = suma_menos_nota_baja / (len(lista2) - 1)
promedio_original = suma_menos_nota_baja / (len(lista2))
print("Promedio CON la nota mas baja", round(promedio_original,0))
print("Promedio sin la nota mas baja", round(promedio, 0))

Nota mas baja: 0
Suma de todas las notas: 69
Suma de todas las notas MENOS la nota mas baja: 69
Promedio CON la nota mas baja 14.0
Promedio sin la nota mas baja 17.0


In [55]:
tupla2 = (18, 12, 0, 20, 19)
nota_baja = min(tupla2)
print("Nota mas baja:", nota_baja)

suma_notas = sum(tupla2)
print("Suma de todas las notas:", suma_notas)
suma_menos_nota_baja1 = suma_notas - nota_baja
print(tupla2.index(nota_baja))
lista2 = list(tupla2)
print(lista2.index(min(lista2)))
lista2.pop(lista2.index(min(lista2)))
print(lista2)


Nota mas baja: 0
Suma de todas las notas: 69
2
2
[18, 12, 20, 19]


# str()

Convertir un numero a string (texto).

En el siguiente ejemplo se intenta concatenar un texto + un numero, lo cual produce un error.

In [80]:
var = 12  # --> "12"
print("El numero es " + var ) # error

TypeError: ignored

In [25]:
print("El numero es " + str(var) )

El numero es 12


# int()

Convierte en int (integer) un numero expresado como texto. Por ejemplo "12".

In [83]:
numero_str = "124"

print( int(numero_str) )

124


In [84]:
num1 = input("Ingresa un numero: ")
print("num es de tipo ", type(num1))

num2 = input("Ingresa un numero: ")
print("num es de tipo ", type(num2))

suma_str = num1 + num2

print("La suma de num1 y num2 es:", num1+num2)

Ingresa un numero: 30
num es de tipo  <class 'str'>
Ingresa un numero: 20
num es de tipo  <class 'str'>
La suma de num1 y num2 es: 3020


In [ ]:
num1 = input("Ingresa un numero: ")

num1_int = int(num1)
print("Ahora, num es de tipo ", type(num1_int))

Ingresa un numero: 123
Ahora, num es de tipo  <class 'int'>


In [ ]:
parte1 = input("Ingresa una nota: ")
parte2 = input("Ingresa otra nota: ")

suma = int(parte1) + int(parte2)
# suma = parte1 + parte2
print("suma es: ",suma)
print("Promedio es: ", suma / 2)

# float()

Convertir a numero real (con parte decimal).

In [86]:
texto_pi = "3.1416"
print("Multiplicar texto_pi por 5: ", texto_pi * 3)

numero_pi = float(texto_pi)
print("Multiplicar numero_pi por 5:", numero_pi * 2)

Multiplicar texto_pi por 5:  3.14163.14163.1416
Multiplicar numero_pi por 5: 6.2832


In [ ]:
parte1 = input("Ingresa una nota: ")
parte2 = input("Ingresa otra nota: ")

suma = float(parte1) + float(parte2)
# suma = parte1 + parte2
print("suma es: ",suma)
print("Promedio es: ", suma / 2)

Ingresa una nota: 14.75
Ingresa otra nota: 10.25
suma es:  25.0
Promedio es:  12.5


# n = 18.734

# m = 15.626

# n= str(n)

# m = str(m)

# n + m


In [91]:
n = 18.734

m = 15.626

n= str(n)

m = str(m)

float(n) + float(m)

34.36

 18.73415.626

 34.36

 18.73 15.62

 error

# input()

Solicita ingreso de datos por teclado.

In [ ]:
nacim = input("Ingrese un numero: ")

print("La funcion input retorna una variable ", type(nacim) )

nacim_int = int(nacim)

print("Con int() la convertimos en entero: ", nacim_int )

Ingrese un numero: 3333
La funcion input retorna una variable  <class 'str'>
Con int() la convertimos en entero:  3333


Funciones que no vienen por default en la instalacion de python se pueden invocar importando la biblioteca o libreria correspondiente.

Por ejemplo, si necesito la funcion de raiz cuadrada (sqrt()). ("squared root")

In [92]:
sqrt(144)

NameError: ignored

In [93]:
import math

In [95]:
print(math.sqrt(144))

12.0


In [97]:
math.cos(90)

math.sin(90)


0.8939966636005579

In [31]:
import math as m

In [32]:
print(m.pi)
print(m.sqrt(144))

3.141592653589793
12.0


Si deseamos obtener el promedio de valores de una lista tenemos que importar *statistics*

In [37]:
import statistics as st

In [98]:
parte1 = (14,	16,	14,	15)
parte2 = (0,	18,	4,	19)
st.median(parte1), st.mean(parte2)

st.mode(parte1)

st.mean(parte1)

14.75

In [ ]:
import statistics as st

In [ ]:
st.mean(parte1), st.mean(parte2)

(14.75, 10.25)